app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala

```diff
@@ -359,7 +352,28 @@ class SectionRenderingService(
       }
       .getOrElse(List(HtmlFormat.empty))
 
-    val snippets = HtmlFormat.fill(infoFields :+ limitReached.fold(addAnotherQuestion)(identity))
+    lazy val radiosWithYes =
+      radios.copy(items = items.copy(head = items.head.copy(checked = true)).toList, fieldset = hiddenFieldset)
+
+    lazy val addAnotherQuestionWithYes: Html =
+      new components.GovukRadios(govukErrorMessage, govukFieldset, govukHint, govukLabel)(radiosWithYes)
+
+    lazy val radiosWithNo =
+      radios.copy(items = items.copy(tail = items.tail.map(_.copy(checked = true))).toList, fieldset = hiddenFieldset)
+
+    lazy val addAnotherQuestionWithNo: Html =
+      new components.GovukRadios(govukErrorMessage, govukFieldset, govukHint, govukLabel)(radiosWithNo)
+
+    val addAnotherQuestionSnippets =
+      (bracket.source.repeatsWhile, bracket.source.repeatsUntil) match {
+        case (Some(_), Some(_)) =>
+          if (evalRepeatsWhile || evalRepeatsUntil) addAnotherQuestionWithNo else addAnotherQuestion
+        case (Some(_), None) => if (evalRepeatsWhile) addAnotherQuestionWithYes else addAnotherQuestion
+        case (None, Some(_)) => if (evalRepeatsUntil) addAnotherQuestionWithNo else addAnotherQuestion
+        case _               => addAnotherQuestion
+      }
```

Comment:
This doesn't make sense. If I understand this correctly it should be simply:

```suggestion
      if (!evalRepeatsWhile)
        addAnotherQuestionWithNo
      else if (!evalRepeatsUntil)
        addAnotherQuestionWithYes
      else addAnotherQuestion
```

Reply:
I think we should check that **repeatsWhile** and **repeatsUntil** has an includeIf before comparison. Because if any of them does not exist, it returns `false` by default. It seems your suggestion does not work for **repeatsWhile** condition.